manager/orchestrator/global/global.go

```diff
@@ -209,8 +209,11 @@ func (g *Orchestrator) handleTaskChange(ctx context.Context, t *api.Task) {
 		return
 	}
 
-	// if a task has passed running, restart it
-	if t.Status.State > api.TaskStateRunning {
+	// Notify the restart supervisor when a task runs¶
+	if t.Status.State == api.TaskStateRunning {
+		g.restarts.Success(t)
```

Comment:
I think this will work properly with health checks. If I remember correctly, the task shouldn't reach the "running" state until the health check passes.

If there is no health check, this won't work very well. The task might reach "running", briefly, and then fail.

I think this is okay, but we'd need to document that a health check is necessary for the backoff feature to work properly. It might even make sense to disallow that setup when no health check is configured.

Reply:
@aaronlehmann i haven't actually dug into the code enough to know if this is feasible, but what if we used logic similar to the update monitor period logic to determine a failure/success?

Reply:
Hi guys, from looking at the updater code I think that using a duration to determine when to reset the failure count would be a fairly straightforward way of doing it and would have the advantage of being independent of health checks.  I don't know enough about the code base to know which would be the "correct" solution though.
Side node:  If we were to go with a monitor duration would using a 5 second default (same as update) seem reasonable.

Reply:
You should use the update monitor period to determine success or failure. If you've passed the update monitor period and the task fails, it's likely a transient blip and you can go ahead and reschedule. If the update monitor period has not yet elapsed, then the failure is likely related to something about the app's bootstrapping and you should back off to wait for the cause of the failure to resolve. Same reason we use this property for updates.